src/ophyd_async/panda/writers/hdf_writer.py

```diff
@@ -0,0 +1,220 @@
+    MinMax = "Min Max"
+    MinMaxMean = "Min Max Mean"
+
+
+def get_capture_signals(
+    panda: Device, path_prefix: Optional[str] = ""
+) -> Dict[str, SignalR]:
+    """Get dict mapping a capture signal's name to the signal itself"""
+    if not path_prefix:
+        path_prefix = ""
+    signals: Dict[str, SignalR[Any]] = {}
+    for attr_name, attr in panda.children():
+        dot_path = f"{path_prefix}{attr_name}"
+        if isinstance(attr, SignalR) and attr_name.endswith("_capture"):
+            signals[dot_path] = attr
+        attr_signals = get_capture_signals(attr, path_prefix=dot_path + ".")
+        signals.update(attr_signals)
+    return signals
+
+
+# This should return a dictionary which contains a dict, containing the Capture
+# signal object, and the value of that signal
+async def get_signals_marked_for_capture(
+    capture_signals: Dict[str, SignalR]
+) -> Dict[str, Dict[str, Union[SignalR, Capture]]]:
+
+    # Read signals to see if they should be captured
+    do_read = [signal.get_value() for signal in capture_signals.values()]
+
+    signal_values = await asyncio.gather(*do_read)
+
+    assert len(signal_values) == len(
+        capture_signals.keys()
+    ), "Length of read signals are different to length of signals"
+    signals_to_capture: Dict[str, Dict[str, Union[SignalR, Capture]]] = {}
+    for signal_path, signal_object, signal_value in zip(
+        capture_signals.keys(), capture_signals.values(), signal_values
+    ):
+
+        signal_path = signal_path.replace("_capture", "")
+        if (signal_value.value in iter(Capture)) and (signal_value.value != Capture.No):
+            signals_to_capture[signal_path] = {
+                "signal": signal_object,
+                "capture_type": signal_value.value,
+            }
+
+    return signals_to_capture
+
+
+class PandaHDFWriter(DetectorWriter):
+    _ctxt: Optional[Context] = None
+
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        name_provider: NameProvider,
+        panda_device: PandA,
+    ) -> None:
+        self.panda_device = panda_device
+        self._prefix = prefix
+        self._directory_provider = directory_provider
+        self._name_provider = name_provider
+        self._datasets: List[_HDFDataset] = []
+        self._file: Optional[_HDFFile] = None
+        self._multiplier = 1
+
+        # Convert the convert the ioc PV names to have a better underscore convention.
+        # Can remove this after Pandablocks issue #101
+        self.hdf = HdfSignals(
+            panda_device.data.hdfdirectory,
+            panda_device.data.hdffilename,
+            panda_device.data.numcapture,
+            panda_device.data.numcaptured,
+            panda_device.data.capture,
+        )
+
+        # Get capture PVs by looking at panda. Gives mapping of dotted attribute path
+        # to Signal object
+        self.capture_signals = get_capture_signals(self.panda_device)
```

Comment:
Unfortunately this is too early, those signals won't exist until after `PandA.connect` so this sound be done in `open` instead